### ISSUE TITLE
Support apply method for NativeFunction

### ISSUE THREAD

**Issue by ChiChou** [frida/frida-gum#150]
NativeFunction does not inherit from Function prototype, so a NativeFunction instance has neither `.apply` nor `.call` method, which are useful when someone wants to call a function whose argument count is dynamic (like `printf`).

P.S.

With the power of eval, a meta programming technique in Javascript world, we can still apply a argument list with variable length to a NativeFunction, using the following hack.

```javascript

function apply(nativeFunc, args) {
  let placeholders = [].slice.call(args).map((e, i) => '_arg' + i);
  let body = 'return f(' + placeholders.join(',') + ')';
  let func = new (Function.bind.apply(Function, [Function, 'f'].concat(placeholders).concat(body)));
  return func.apply(null, [nativeFunc].concat(args));
}

// call the function
var fd = apply(open, [Memory.allocUtf8String('/tmp/test.txt'), 256]);
```

**Comment by riverar** [comment]
Haha, I was literally working on this then abandoned it because I didn't see a lot of use. Might have to revisit.

**Comment by oleavr** [comment]
Implemented in #177 and available as of 8.1.3, released last night.

**Comment by ChiChou** [comment]
Awesome work! Thanks
